services/mirror/mirror_pull.go

```diff
@@ -215,13 +216,21 @@ func runSync(ctx context.Context, m *repo_model.Mirror) ([]*mirrorSyncResult, bo
 		return nil, false
 	}
 
+	envs := []string{}
+	if strings.EqualFold(remoteURL.Scheme, "http") || strings.EqualFold(remoteURL.Scheme, "https") {
+		if proxy.Match(remoteURL.Host) {
+			envs = append(envs, fmt.Sprintf("https_proxy=%s", proxy.GetProxyURL()))
+		}
+	}
+
```

Comment:
Alternatively, instead of duplicating the code, what about using a common function instead?